syft/pkg/cataloger/dotnet/parse_dotnet_portable_executable.go

```diff
@@ -0,0 +1,85 @@
+package dotnet
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/saferwall/pe"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+var _ generic.Parser = parseDotnetPortableExecutable
+
+func parseDotnetPortableExecutable(_ file.Resolver, _ *generic.Environment, f file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	by, err := io.ReadAll(f)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to read file: %w", err)
+	}
+
+	peFile, err := pe.NewBytes(by, &pe.Options{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to create PE file instance: %w", err)
+	}
+
+	err = peFile.Parse()
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to parse PE file: %w", err)
+	}
+
+	versionResources, err := peFile.ParseVersionResources()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	name := versionResources["FileDescription"]
+	if name == "" {
+		log.Debugf("unable to find FileDescription in PE file: %s", f.RealPath)
+		return nil, nil, nil
+	}
+
+	version := versionResources["FileVersion"]
+	if version == "" {
+		log.Debugf("unable to find FileVersion in PE file: %s", f.RealPath)
+		return nil, nil, nil
+	}
+
+	purl := packageurl.NewPackageURL(
+		packageurl.TypeNuget, // See explanation in syft/pkg/cataloger/dotnet/package.go as to why this was chosen.
+		"",
+		name,
+		version,
+		nil,
+		"",
+	).ToString()
+
+	metadata := pkg.DotnetPortableExecutableMetadata{
+		AssemblyVersion: versionResources["Assembly Version"],
+		LegalCopyright:  versionResources["LegalCopyright"],
+		Comments:        versionResources["Comments"],
+		InternalName:    versionResources["InternalName"],
+		CompanyName:     versionResources["CompanyName"],
+		ProductName:     versionResources["ProductName"],
+		ProductVersion:  versionResources["ProductVersion"],
+	}
+
+	p := pkg.Package{
+		Name:         name,
+		Version:      version,
+		Locations:    file.NewLocationSet(f.Location),
+		Language:     pkg.Dotnet,
```

Comment:
Is there something in particular about how the version resources are built that we can assume this is `.net`? The cataloger is selecting files that end in .dll and .exe, which don't guarantee this is written in a `.net` language.

Reply:
Good question, and I'm thinking you might be right. Incidentally my first draft left this field empty, which seems like a more correct representation of what's here. I'll make that adjustment.

Reply:
( --> https://github.com/anchore/syft/pull/1943/commits/368313fddac9160d8a06a01ebe8c5ac7990232f5 )